src/store/reducers.ts

```diff
@@ -8,50 +11,60 @@ import {
   DELETE_HABIT,
   UPDATE_STREAK,
   UPDATE_CURRENT_USER,
-  REGISTER_USER
-} from "../actions";
+  REGISTER_USER,
+  State,
+  ActionTypes,
+  UserState,
+} from "./types";
 
-const initialState = {
+const initialState: State = {
   initialized: false, // false means the state hasn't been read from localstorage yet
   currentUser: null, // Null means no one is logged in, a string identifies a username
-  userStates: {}
+  userStates: {},
 };
 
 const STATE_KEY = "state";
 
 // Write redux state to local storage
-export const writeStateMiddleware = store => next => action => {
+export const writeStateMiddleware: Middleware<{}, State> = (store) => (
+  next
+) => (action: ActionTypes) => {
   let result = next(action);
   const { initialized, ...state } = store.getState();
   window.localStorage.setItem(STATE_KEY, JSON.stringify(state));
   return result;
 };
 
 // Retrieve Redux state from local storage
-export const readState = () => {
+export const readState = (): State => {
   try {
     const stateJson = window.localStorage.getItem(STATE_KEY);
```

Comment:
This could be another followup -- it doesn't need to stop this PR from being merged -- but we really ought to use the safe `getLocalStorageItem` function here and add `STATE_KEY` to the `LocalStorageKey` type. Functionally its the same, but we get a little more consistency and safety from that.